### DISCUSSION TITLE
Memory usage in ESP32-C3-DevKitM-1

### DISCUSSION THREAD

**Discussion by pablogventura** [micropython/micropython#12322] [ESP32]
I am building for an ESP32-C3-DevKitM-1, choosing GENERIC_C3 as BOARD at compile time. The output of the compilation is:

```Project build complete. To flash, run this command:
/home/pablo/.espressif/python_env/idf4.4_py3.10_env/bin/python ../../../../esp-idf/components/esptool_py/esptool/esptool.py -p (PORT) -b 460800 --before default_reset --after hard_reset --chip esp32c3  write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x0 build-GENERIC_C3/bootloader/bootloader.bin 0x8000 build-GENERIC_C3/partition_table/partition-table.bin 0x10000 build-GENERIC_C3/micropython.bin
or run 'idf.py -p (PORT) flash'

bootloader @0x000000    16128  (   16640 remaining)

partitions  @0x008000     3072  (    1024 remaining)

application @0x010000  1412480  (  619136 remaining)

total                  1478016
```

So I don't understand the remaining memory since it doesn't seem to reach the 4MB that the ESP32-C3-MINI-1 has. Where is configured how much memory flash has in MicroPython? How is the message in the output of make interpreted?

**Comment by jimmo** [comment]
This output is listing the usage of the partitions in the firmware. See https://github.com/micropython/micropython/blob/master/ports/esp32/partitions-4MiB.csv

Notably, the "vfs" partition (which is 2MiB) is not included here (as it's not included in the firmware image as it needs to persist across firmware updates).

So in summary: you have ~1.3 MiB of code (leaving 604 kiB) of the roughly 2MiB of flash used for code. The remaining 2MiB is used for the filesystem.

**Comment by pablogventura** [comment]
Thank you very much for your answer! If I add modules in c or frozen Python code, could I get to occupy more leaving the filesystem with less than 2mb?

**Reply by jimmo** [reply]
@pablogventura I'm not sure what you mean sorry.

**Reply by pablogventura** [reply]
Sorry, I didn't make myself clear. My intention is to add C components to my own MicroPython build, and I wanted to know if I could decrease the size of the VFS partition to get more space for MicroPython and the C components I add.
